engines/data_export/lib/data_export/engine.rb

```diff
@@ -30,14 +30,49 @@ def export_rmt_data
         after_action :export_rmt_data, only: %i[update], if: -> { response.successful? && !@system.byos? }
 
         def export_rmt_data
+          if @system.products.empty?
+            send_data
```

Comment:
I don't think we want to send anything if there is no product data. I don't quite see how data without the association to a given product would be useful.

Reply:
only sending the data when systems controller performs an update if the system has products fixed in https://github.com/SUSE/rmt/pull/1322/commits/1f5fc18f1942ae85a63e47a84d29319120e7cd3b